executor/index_merge_reader.go

```diff
@@ -320,6 +307,25 @@ func (w *partialTableWorker) fetchHandles(ctx context.Context, fetchCh chan<- *l
 	}
 }
 
+func (w *partialTableWorker) handleHandlesFetcherPanic(ctx context.Context, resultCh chan<- *lookupTableTask, err error) func(r interface{}) {
+	return func(r interface{}) {
+		if r == nil {
+			return
+		}
+
+		logutil.Logger(ctx).Error("partialTableWorker in IndexMergeReaderExecutor panicked")
+		err4Panic := errors.Errorf("%v", r)
+		doneCh := make(chan error, 1)
+		doneCh <- err4Panic
+		resultCh <- &lookupTableTask{
+			doneCh: doneCh,
+		}
+		if err != nil {
```

Comment:
should it be `if err == nil`?

Reply:
I made a mistake...
`if (err != nil)  err = errors.Trace(err4Panic)` will add `err4Panic` in `error`.
and `error` with `err4Panic` will be printed by `logutil.Logger(ctx).Error("close Select result failed:", zap.Error(err))`.

so ... `if err != nil {` in there is senseless。
may be I should return error in `worker.handleHandlesFetcherPanic(ctx, e.resultCh, err)`

Reply:
`if err != nil { err = errors.Trace(err4Panic) }`
can I remove this code in there?
It seems to be meaningless。

Reply:
OK